cypress/integration/inventory/Item-status-date-updates.spec.js

```diff
@@ -0,0 +1,376 @@
+      permissions.uiInventoryStorageModule.gui,
+      permissions.uiUsersDeclareItemLost.gui,
+      permissions.usersLoansRenewThroughOverride.gui
+    ])
+      .then(userProperties => {
+        user = userProperties;
+        cy.getAdminToken()
+          .then(() => {
+            cy.getOrganizationApi({ query: 'name="Amazon.com"' })
+              .then(organization => {
+                order.vendor = organization.id;
+                orderLine.physical.materialSupplier = organization.id;
+                orderLine.eresource.accessProvider = organization.id;
+              });
+            cy.getMaterialTypes({ query: 'name="book"' })
+              .then(materialType => { orderLine.physical.materialType = materialType.id; });
+          });
+        effectiveLocationServicePoint = NewServicePoint.getDefaulServicePoint();
+        effectiveLocationServicePointName = effectiveLocationServicePoint.body.name;
+        const effectiveLocationServicePointId = effectiveLocationServicePoint.body.id;
+        notEffectiveLocationServicePoint = NewServicePoint.getDefaulServicePoint();
+        notEffectiveLocationServicePointName = notEffectiveLocationServicePoint.body.name;
+
+        Institutions.createViaApi(Institutions.defaultUiInstitutions.body);
+        const specialCampuse = { ...Campuses.defaultUiCampuses.body };
+        specialCampuse.institutionId = Institutions.defaultUiInstitutions.body.id;
+        Campuses.createViaApi(specialCampuse);
+        const specialLibrary = { ...Libraries.defaultUiLibraries.body };
+        Libraries.createViaApi(specialLibrary);
+        specialLibrary.campusId = specialCampuse.id;
+
+        effectiveLocation = { ...NewLocation.getDefaultUiLocation(Institutions.defaultUiInstitutions.body.id, specialCampuse.id, specialLibrary.id).body, servicePointIds: [effectiveLocationServicePoint.body.id], primaryServicePoint: effectiveLocationServicePoint.body.id };
+
+        userRequestPreferences = {
+          id: uuid(),
+          fulfillment: 'Delivery',
+          defaultDeliveryAddressTypeId: null,
+          defaultServicePointId: effectiveLocationServicePointId,
+          delivery: true,
+          holdShelf: true,
+          userId: userForDeliveryRequest.id,
+        };
+
+        ServicePoints.createViaApi(effectiveLocationServicePoint.body);
+        ServicePoints.createViaApi(notEffectiveLocationServicePoint.body);
+        cy.addServicePointToUser([effectiveLocationServicePoint.body.id, notEffectiveLocationServicePoint.body.id],
+          user.userId, effectiveLocationServicePoint.body.id);
+        Locations.createLocationViaApi(effectiveLocation)
+          .then(locations => {
+            orderLine.locations[0].locationId = locations.body.id;
+            itemLocation = locations.body.name;
+          });
+        cy.login(userProperties.username, userProperties.password)
+          .then(() => {
+            Orders.createOrderWithOrderLineViaApi(order, orderLine)
```

Comment:
you can create getOrderWithVendor(vendorId) and use it instead of management of order into the test. Additionally, if vendorId is undefined you can run adding of vendor into this order getter

Reply:
Done